backend/src/main/java/io/debezium/configserver/service/mongodb/MongoDbConnectorIntegrator.java

```diff
@@ -99,9 +96,9 @@ public class MongoDbConnectorIntegrator extends ConnectorIntegratorBase {
         MONGODB_PROPERTIES =  Collections.unmodifiableMap(additionalMetadata);
     }
 
-    protected ConnectionContext.MongoPrimary primary(MongoDbTaskContext context) throws Throwable {
-        ReplicaSet replicaSet = ReplicaSet.parse(context.getConnectionContext().hosts());
-        return context.getConnectionContext().primaryFor(replicaSet, context.filters(), (s, throwable) -> {
+    protected MongoDbConnection primary(MongoDbTaskContext context) throws Throwable {
+        ReplicaSet replicaSet = new ReplicaSet(context.getConnectionContext().connectionString());
```

Comment:
I think the test is failing as the new Mongo DB instance is using a mapped port which needs to be fetched, could that be? Not entirely sure.